ddtrace/contrib/internal/openai_agents/utils.py

```diff
@@ -0,0 +1,148 @@
+        manifest["handoffs"] = handoffs
+
+    guardrails = extract_guardrails_from_agent(agent)
+    if guardrails:
+        manifest["guardrails"] = guardrails
+
+    return manifest
+
+
+def extract_model_settings_from_agent(agent):
+    if not hasattr(agent, "model_settings"):
+        return None
+
+    # convert model_settings to dict if it's not already
+    model_settings = agent.model_settings
+    if type(model_settings) != dict:
+        if hasattr(model_settings, "__dict__"):
+            model_settings = model_settings.__dict__
+        else:
+            return None
+
+    return make_json_compatible(model_settings)
+
+
+def extract_tools_from_agent(agent):
+    if not hasattr(agent, "tools"):
+        return None
+
+    tools = []
+    for tool in agent.tools:
+        tool_dict = {}
+        if isinstance(tool, WebSearchTool):
+            if hasattr(tool, "user_location"):
+                tool_dict["user_location"] = tool.user_location
+            if hasattr(tool, "search_context_size"):
+                tool_dict["search_context_size"] = tool.search_context_size
+        elif isinstance(tool, FileSearchTool):
+            if hasattr(tool, "vector_store_ids"):
+                tool_dict["vector_store_ids"] = tool.vector_store_ids
+            if hasattr(tool, "max_num_results"):
+                tool_dict["max_num_results"] = tool.max_num_results
+            if hasattr(tool, "include_search_results"):
+                tool_dict["include_search_results"] = tool.include_search_results
+        elif isinstance(tool, ComputerTool):
+            if hasattr(tool, "name"):
+                tool_dict["name"] = tool.name
+        else:
+            if hasattr(tool, "name"):
+                tool_dict["name"] = tool.name
+            if hasattr(tool, "description"):
+                tool_dict["description"] = tool.description
+            if hasattr(tool, "strict_json_schema"):
+                tool_dict["strict_json_schema"] = tool.strict_json_schema
+            if hasattr(tool, "params_json_schema"):
+                parameter_schema = tool.params_json_schema
+                required_params = {param: True for param in parameter_schema.get("required", [])}
+                parameters = {}
+                for param, schema in parameter_schema.get("properties", {}).items():
+                    param_dict = {}
+                    if "type" in schema:
+                        param_dict["type"] = schema["type"]
+                    if "title" in schema:
+                        param_dict["title"] = schema["title"]
+                    if param in required_params:
+                        param_dict["required"] = True
+                    parameters[param] = param_dict
+                tool_dict["parameters"] = parameters
+        tools.append(tool_dict)
+
+    return tools
+
+
+def extract_handoffs_from_agent(agent):
+    if not hasattr(agent, "handoffs"):
+        return None
+
+    handoffs = []
+    for handoff in agent.handoffs:
```

Comment:
```suggestion
    for handoff in getattr(agent, "handoffs", []):
```
probably less likely than tools but could `handoffs` ever be `None`?

Reply:
Same thing as for `agent.tools` above but lmk if you think I should change this!